mender.go

```diff
@@ -15,28 +15,30 @@ package main
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"os"
 	"strings"
 	"time"
 
 	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 type Controler interface {
-	Bootstrap() error
-	TransitionState() MenderState
+	Bootstrap() menderError
 	GetCurrentImageID() string
-	GetDaemonConfig() daemonConfig
-	GetUpdaterConfig() httpsClientConfig
-	LastError() error
+	GetUpdatePollInterval() time.Duration
+	HasUpgrade() (bool, menderError)
+	CheckUpdate() (*UpdateResponse, menderError)
```

Comment:
Might be that we have different views on what Controller controls. I originally assumed that the controller will be driving the update check/download/install/apply/reboot steps, becoming a device/update controller of sorts.

For instance, I expect the reporting part to be separate from device controller, with only some loose coupling through a channel or callbacks. Similarly the inventory piece, it's kind of outside of the update cycle.

The central point where all of these come together would be the daemon code.